cmd/krel/cmd/release_notes.go

```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
 		tag = releaseNotesOpts.tag
 	}
 
-	s, err := util.TagStringToSemver(tag)
-	if err != nil {
-		return errors.Wrapf(err, "no valid tag: %v", tag)
+	// First, validate cmdline options
+	if err := releaseNotesOpts.Validate(); err != nil {
+		return errors.Wrap(err, "validating command line options")
 	}
-	start := util.SemverToTagString(semver.Version{
-		Major: s.Major,
-		Minor: s.Minor - 1,
-		Patch: 0,
-		Pre:   []semver.PRVersion{{VersionStr: "rc.1"}},
-	})
-
-	var result *releaseNotesResult
 
-	// Create the PR for relnotes.k8s.io
+	// before running the generators, verify that the repositories are ready
 	if releaseNotesOpts.createWebsitePR {
-		// Check cmd line options
-		if err = validateWebsitePROptions(); err != nil {
-			return errors.Wrap(err, "validating PR command line options")
+		if err = verifyFork(
+			fmt.Sprintf("%s%s", websiteBranchPrefix, tag),
+			releaseNotesOpts.GithubOrg, releaseNotesOpts.websiteRepo,
+			defaultKubernetesSigsOrg, defaultKubernetesSigsRepo,
+		); err != nil {
+			return errors.Wrapf(err, "while checking %s/%s fork", defaultKubernetesSigsOrg, defaultKubernetesSigsRepo)
 		}
+	}
 
-		// Verify the repository
-		if err = validateWebsitePRRepository(); err != nil {
+	if releaseNotesOpts.createDraftPR {
+		if err = verifyFork(
+			fmt.Sprintf("%s%s", draftBranchPrefix, tag),
+			releaseNotesOpts.GithubOrg, releaseNotesOpts.draftRepo,
+			git.DefaultGithubOrg, git.DefaultGithubReleaseRepo,
+		); err != nil {
 			return errors.Wrapf(err, "while checking %s/%s fork", defaultKubernetesSigsOrg, defaultKubernetesSigsRepo)
 		}
+	}
 
-		// Generate the release notes for ust the current tag
-		jsonStr, err := releaseNotesJSON(tag)
-		if err != nil {
-			return errors.Wrapf(err, "generating release notes")
-		}
+	var websiteError, draftError error
 
+	// Create the PR for relnotes.k8s.io
+	if releaseNotesOpts.createWebsitePR {
 		// Run the website PR process
-		if err := createWebsitePR(tag, jsonStr); err != nil {
-			return errors.Wrapf(err, "generating release notes for tag %s", tag)
-		}
-		return nil
+		websiteError = createWebsitePR(tag)
```

Comment:
Is there any reason that we do not return earlier in case of an error, like:
```suggestion
		return errors.Wrap(createWebsitePR(tag), "creating website PR")
```